packages/dropdowns/src/styled/menu/StyledMenu.tsx

```diff
@@ -34,88 +76,124 @@ const retrieveMenuMargin = ({
+  `;
 };
 
 interface IStyledMenuViewProps extends HTMLAttributes<HTMLUListElement> {
   isCompact?: boolean;
-  placement?: POPPER_PLACEMENT;
   isAnimated?: boolean;
-  hasArrow?: boolean;
   isHidden?: boolean;
+  hasArrow?: boolean;
+  placement?: POPPER_PLACEMENT;
 }
 
 /**
- * Accepts all `<ul>` props
+ * 1. Positioned relative to controlling item.
+ * 2. Opt out of browser default list margin.
+ * 3. Prevent controlling item cursor inheritance.
+ * 4. Opt out of browser default list padding.
+ * 5. Prevent controlling item whitespace inheritance.
  */
 const StyledMenuView = styled.ul.attrs<IStyledMenuViewProps>(props => ({
   'data-garden-id': COMPONENT_ID,
   'data-garden-version': PACKAGE_VERSION,
-  className: classNames(MenuStyles['c-menu'], {
-    // Size
-    [MenuStyles['c-menu--sm']]: props.isCompact,
-
-    // Placement
-    [MenuStyles['c-menu--up']]:
-      props.placement === 'top' || props.placement === 'top-start' || props.placement === 'top-end',
-    [MenuStyles['c-menu--right']]:
-      props.placement === 'right' ||
-      props.placement === 'right-start' ||
-      props.placement === 'right-end',
-    [MenuStyles['c-menu--left']]:
-      props.placement === 'left' ||
-      props.placement === 'left-start' ||
-      props.placement === 'left-end',
-    [MenuStyles['c-menu--down']]:
-      props.placement === 'bottom' ||
-      props.placement === 'bottom-start' ||
-      props.placement === 'bottom-end',
-
-    // State
-    [MenuStyles['is-open']]: props.isAnimated,
-    [MenuStyles['is-hidden']]: props.isHidden,
-
-    // Arrows
-    [ArrowStyles['c-arrow']]: shouldShowArrow({
-      hasArrow: props.hasArrow,
-      placement: props.placement
-    }),
-    [ArrowStyles['c-arrow--r']]: props.placement === 'left',
-    [ArrowStyles['c-arrow--rt']]: props.placement === 'left-start',
-    [ArrowStyles['c-arrow--rb']]: props.placement === 'left-end',
-    [ArrowStyles['c-arrow--b']]: props.placement === 'top',
-    [ArrowStyles['c-arrow--bl']]: props.placement === 'top-start',
-    [ArrowStyles['c-arrow--br']]: props.placement === 'top-end',
-    [ArrowStyles['c-arrow--l']]: props.placement === 'right',
-    [ArrowStyles['c-arrow--lt']]: props.placement === 'right-start',
-    [ArrowStyles['c-arrow--lb']]: props.placement === 'right-end',
-    [ArrowStyles['c-arrow--t']]: props.placement === 'bottom',
-    [ArrowStyles['c-arrow--tl']]: props.placement === 'bottom-start',
-    [ArrowStyles['c-arrow--tr']]: props.placement === 'bottom-end',
-
-    // RTL
-    [MenuStyles['is-rtl']]: isRtl(props)
-  })
+  className: props.isAnimated && 'is-animated'
 }))<IStyledMenuViewProps>`
-  && {
-    position: relative;
-  }
+  display: ${props => props.isHidden && 'inline-block'};
+  position: relative; /* [1] */
+  transition: ${props => props.isHidden && 'opacity .2s ease-in-out, .2s visibility 0s linear'};
+  visibility: ${props => props.isHidden && 'hidden'};
+  opacity: ${props => props.isHidden && '0'};
```

Comment:
This was already broken in `master`, but I was able to correct all of this logic and also improve our `top|right|left` positioning.

In the original `css-components` the `is-hidden` class had to be applied to the menu element. Since we are using `::before/::after` arrow styling I've moved the `isHidden` logic to the element which wraps the menu (and actually controls PopperJS). This seems to have fixed a good chunk of the issues.